app/controllers/api/base_controller.rb

```diff
@@ -276,7 +276,7 @@ def parent_permission(child_permission)
       case child_permission.to_s
       when 'create', 'destroy'
         'edit'
-      when 'edit', 'view'
+      when 'edit', 'view', 'import_puppetclasses'
```

Comment:
IIUC, it appears that since the environment controller is called from within the smart proxy controller, it attempts to map which permission is required for the parent controller. Since it does not know how to handle the `import_puppetclasses` action here, just adding it in the environment controller caused this to throw an exception. I will try to move it out to the environment controller.